Source/Request.swift

```diff
@@ -414,3 +323,283 @@ extension Request: CustomDebugStringConvertible {
         return components.joined(separator: " \\\n\t")
     }
 }
+
+// MARK: -
+
+/// Specific type of `Request` that manages an underlying `URLSessionDataTask`.
+open class DataRequest: Request {
+
+    // MARK: Helper Types
+
+    enum Requestable: TaskConvertible {
+        case request(URLRequest)
+
+        func task(session: URLSession, adapter: RequestAdapter?, queue: DispatchQueue) -> URLSessionTask {
+            var task: URLSessionTask!
+
+            switch self {
+            case let .request(urlRequest):
+                let urlRequest = urlRequest.adapt(using: adapter)
+                queue.sync { task = session.dataTask(with: urlRequest) }
```

Comment:
Added `syncResult` in b56bd7d8. I changed it in most places throughout the codebase except in places that I know were changed in other PRs that will generate merge conflicts.